include/zephyr/bluetooth/iso.h

```diff
@@ -621,10 +621,121 @@ int bt_iso_chan_disconnect(struct bt_iso_chan *chan);
  */
 int bt_iso_chan_send(struct bt_iso_chan *chan, struct net_buf *buf);
 
+struct bt_iso_unicast_tx_info {
+	/** The transport latency in us */
+	uint32_t latency;
+
+	/** The flush timeout in ms */
+	uint32_t flush_timeout;
+
+	/** The transport PHY  */
+	uint8_t  phy;
+
+	/** The burst number */
+	uint8_t  bn;
+
+	/** The maximum PDU size in octets */
+	uint16_t max_pdu;
+};
+
+struct bt_iso_unicast_info {
+	/** The maximum time in us for all PDUs of all CIS in a CIG event */
+	uint32_t cig_sync_delay;
+
+	/** The maximum time in us for all PDUs of this CIS in a CIG event */
+	uint32_t cis_sync_delay;
+
+	/** The maximum number of subevents in each ISO event */
+	uint8_t  max_subevent;
+
+	/** The ISO interval in ms */
+	uint16_t interval;
+
+	/** @brief TX information when sending from central to peripheral
+	 *
+	 *  All values will be 0 if the central cannot send to the peripheral.
+	 */
+	struct bt_iso_unicast_tx_info central;
+
+	/** TX information when sending from peripheral to central
+	 *
+	 *  All values will be 0 if the peripheral cannot send to the central.
+	 */
+	struct bt_iso_unicast_tx_info peripheral;
+};
+
+struct bt_iso_broadcaster_info {
+	/** The maximum time in us for all PDUs of all BIS in a BIG event */
+	uint32_t sync_delay;
+
+	/** The transport latency in us */
+	uint32_t latency;
+
+	/** The transport PHY  */
+	uint8_t  phy;
+
+	/** The maximum number of subevents in each ISO event */
+	uint8_t  max_subevent;
+
+	/** The burst number */
+	uint8_t  bn;
+
+	/** Number of times a payload is transmitted in a BIS event */
+	uint8_t  irc;
+
+	/** Pre-transmission offset in ms */
+	uint32_t  pto;
+
+	/** The maximum PDU size in octets */
+	uint16_t max_pdu;
+
+	/** The ISO interval in ms */
+	uint16_t interval;
+};
+
+struct bt_iso_sync_receiver_info {
```

Comment:
This is the same as `bt_iso_broadcaster_info` just without the `sync_delay`. Should we use the same struct for both, and then just document that `sync_delay` is for the broadcaster only?

Reply:
From an application data path point of view there is no difference between a broadcast and a unicast. Perhaps it would make sense to have the data common between unicast and broadcast in one structure (BN, PHY, max_pdu, number_of_subevents and interval) to have a generic data path handler - but there is also value in having simple structures with only valid members. Even if the documentation mentions a value is only valid in some cases a developer might miss that - especially during maintenance where you only modify the code.

Reply:
The difference between unicast and broadcast is taht unicast may have 2 pairs of latency, flush timeout, etc. as it can be bidirectional.